pkg/network/compress.go

```diff
@@ -1,33 +1,29 @@
 package network
 
 import (
-	"bytes"
-	"io"
-
 	"github.com/pierrec/lz4"
 )
 
 // compress compresses bytes using lz4.
 func compress(source []byte) ([]byte, error) {
-	dest := new(bytes.Buffer)
-	w := lz4.NewWriter(dest)
-	_, err := io.Copy(w, bytes.NewReader(source))
+	dest := make([]byte, lz4.CompressBlockBound(len(source)))
+	size, err := lz4.CompressBlock(source, dest, nil)
 	if err != nil {
 		return nil, err
 	}
-	if w.Close() != nil {
-		return nil, err
-	}
-	return dest.Bytes(), nil
+	return dest[:size], nil
 }
 
 // decompress decompresses bytes using lz4.
 func decompress(source []byte) ([]byte, error) {
-	dest := new(bytes.Buffer)
-	r := lz4.NewReader(bytes.NewReader(source))
-	_, err := io.Copy(dest, r)
+	maxSize := len(source) * 255
```

Comment:
Where does this 255 came from?

Reply:
From the C# code. Yeah, I don't like it too. But as we don't have a real uncompressed size we have two basic options:
 * always allocate the maximum allowed payload size (and it's biggy)
 * try to apply some heuristics, that's what is being done here and it's quite safe

But at the same time I think adding 4 bytes into the compressed payload to specify the length would be so much better... Maybe we'll make a proposal, but for now we have what we have.